### PR TITLE
Initial commit of working datasets docker cicd

### PR DESCRIPTION
**Pull request by ens-LCampbell**
Continuation of testing PR to integrate automatic generation of docker images for datasets cli.

What remains:
- How to schedule container building via time interval given we have no more scheduling slots available on gitlab. 
- Alternatively we could implement a manual trigger to initiate the container build.

NB: For now this PR can remain un merged until a schedule can be established.

### CODE REVIEW

File: cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# License-related part of the Ensembl/ensembl-genomio CI/CD pipeline
+
+# A generic job to be used by all License jobs with the "extends" mechanism
+# (https://docs.gitlab.com/ee/ci/yaml/#extends)
+
+
+variables:
+  # https://docs.gitlab.com/ee/user/namespace/  
+  CONTAINER_IMAGE_NAME: datasets-cli
+  DATASETS_VERSION: "16.20.0"
```
Comment (ens-LCampbell) [inline cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml:24]: This variable is required to be updated manually when new datasets cli versions are released. At the moment, a separate cron job would check this, and email metazoan email address with a prompt to update it. Outside of this option, its tricky to obtain when checking via curl as gitlab runners don't have curl by default

**Comment by JAlvarezJarreta** [comment]
Leaving this note here as agreed in the soft. dev. meeting this morning: this container and its pipeline is meant to be moved to the ncbi-datasets repository once made public.

**Review by MatBarba — COMMENTED** [review]
Looks good, with a couple comments.

For the gitlab yml file, are the duplicated "echo {command}" parts needed?

File: cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - docker:dind
+
+  tags:
+    - dind
+  
+  script:
+    # Get the latest version of NCBI's datasets-cli software
+    # https://docs.gitlab.com/ee/ci/variables/predefined_variables.html    
+    - echo "Building $CONTAINER_IMAGE_NAME"
+    - echo "docker build -f ${DATASETS_DIR}/Dockerfile --build-arg datasets_version=$DATASETS_VERSION -t $CONTAINER_IMAGE_NAME $DATASETS_DIR"
```
Comment (vsitnik) [inline cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml:53]: I wonder if it's possible to get url and version with, store them in the files
with something like
```
wget https://api.github.com/repos/ncbi/datasets/releases/latest -O - | grep browser_download_url | grep linux-amd64.cli.package.zip |  cut -f 4 -d '"' | tee container_url.txt | cut -f 8 -d / > container_ver.txt
```
and  use them to build the image or publish it

**Review by vsitnik — APPROVED** [review]

File: cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml (hunk 1 above)
Comment (ens-LCampbell) [reply cicd/gitlab/parts/build-datasets-dual-deploy.gitlab-ci.yml]: Recent updates to cicd docker yml no longer requires manual intervention to grab and embed the most up to date version. In the end to get this working it required a new stage running ubuntu docker image and passing of variables between jobs using artifacts:reports:dotenv

**Pull request closed without merging**